config.go

```diff
@@ -1542,6 +1546,35 @@ func (c *Config) ImplementationConfig() ImplementationCfg {
 		},
 	}
 
+	// If we're using a remote signer, we still need the base wallet as a
+	// watch-only source of chain and address data. But we don't need any
+	// private key material in that btcwallet base wallet.
+	if c.RemoteSigner.Enable {
```

Comment:
So I think this is close, but it should still be one "level" higher. So rather than us checking this portion of the config here, we'd instead do it within `cmd/lnd`. The difference is that in that instances, `lnd` doesn't need to handle the initialization, and instead just accepts the fully initialized interfaces. This way, things can evolve outside of lnd (via wrapper binaries kinda like `litd`) without needing to change anything on the bsae `lnd` level. 

In practice, this means more config parsing being moved into `cmd/lnd`.